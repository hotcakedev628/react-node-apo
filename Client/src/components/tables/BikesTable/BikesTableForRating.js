import React from 'react'
import { BasicTableHeaders, BasicTableDescription } from './BikesTableFragments';
import { BikesTableLayout } from './BikesTableLayout';
import StarRatingComponent from 'react-star-rating-component';


export const BikesTableForRating = ({ bikesDetails, onRateClick }) => (
    <BikesTableLayout
        TableHeaders={() =>
            <React.Fragment>
                <BasicTableHeaders />
                <th>Number of Rides</th>
                <th className="text-center">Rate</th>
            </React.Fragment>
        }

        TableBodyContent={() =>
            bikesDetails.map((item, index) =>
                (
                    <tr key={index}>
                        <BasicTableDescription item={item.bike} />
                        <td>{item.numberOfRides}</td>


                        <td className="text-center ">
                            <StarRatingComponent
                                name={item.bike._id} /* name of the radio input, it is required */
                                value={item.rate || 0 } /* number of selected icon (`0` - none, `1` - first) */
                                starCount={5} /* number of icons in rating, default `5` */
                                onStarClick={onRateClick} /* on icon click handler */
                            />
                        </td>
                    </tr>
                )
            )
        }
    />
)

