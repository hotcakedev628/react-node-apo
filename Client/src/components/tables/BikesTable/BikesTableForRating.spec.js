import renderer from 'react-test-renderer';
import {BikesTableForRating} from './BikesTableForRating';
import React from 'react'



describe('BikesTableForRating', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<BikesTableForRating 
        bikesDetails={[{rate: 4, bike:{model: 'as', color: 'red', _id: 'aa'}}]}
        areReservationsAllowed = {true}
        onAddClick = {()=>{}}
        onEditClick = {()=>{}}
        />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
