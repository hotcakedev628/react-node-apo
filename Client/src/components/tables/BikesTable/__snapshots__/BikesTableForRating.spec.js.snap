// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BikesTableForRating should match snapshot 1`] = `
<div>
  <table
    className="table table-hover"
    id="BikesTable"
  >
    <thead>
      <tr>
        <th>
          Image
        </th>
        <th>
          Model
        </th>
        <th>
          Color
        </th>
        <th>
          Weight
        </th>
        <th>
          Number of Rides
        </th>
        <th
          className="text-center"
        >
          Rate
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <p>
            No image available
          </p>
        </td>
        <td>
          as
        </td>
        <td>
          red
        </td>
        <td />
        <td />
        <td>
          <div
            className="dv-star-rating"
            style={
              Object {
                "display": "inline-block",
                "position": "relative",
              }
            }
          >
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_5"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={5}
            />
            <label
              className="dv-star-rating-star dv-star-rating-empty-star"
              htmlFor="aa_5"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#333",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={true}
              className="dv-star-rating-input"
              id="aa_4"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={4}
            />
            <label
              className="dv-star-rating-star dv-star-rating-full-star"
              htmlFor="aa_4"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#ffb400",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_3"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={3}
            />
            <label
              className="dv-star-rating-star dv-star-rating-full-star"
              htmlFor="aa_3"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#ffb400",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_2"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={2}
            />
            <label
              className="dv-star-rating-star dv-star-rating-full-star"
              htmlFor="aa_2"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#ffb400",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
            <input
              checked={false}
              className="dv-star-rating-input"
              id="aa_1"
              name="aa"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                  "marginLeft": -9999,
                  "position": "absolute",
                }
              }
              type="radio"
              value={1}
            />
            <label
              className="dv-star-rating-star dv-star-rating-full-star"
              htmlFor="aa_1"
              onClick={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "color": "#ffb400",
                  "cursor": "pointer",
                  "float": "right",
                }
              }
            >
              <i
                style={
                  Object {
                    "fontStyle": "normal",
                  }
                }
              >
                ★
              </i>
            </label>
          </div>
          <p>
            4
             
            <span>
                Stars  
            </span>
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
